### PR TITLE
WIP: Add new events added in Lucee Hibernate extension v5.4

### PR DESCRIPTION
**Pull request by michaelborn**
New events for Lucee's Hibernate extension version 5.4:

* `onDirtyCheck` :sponge: 
* `onEvict` :key: 
* `onClear` :wastebasket: 
* `onFlush` :toilet: 
* `onAutoFlush` :roll_of_paper: 

I could use help writing tests for these! :smile: I see [one event test in BaseORMServiceTest.cfc](https://github.com/coldbox-modules/cborm/blob/bfd645520f868be98d4d4a07a85bb26571ab847a/test-harness/tests/specs/BaseORMServiceTest.cfc#L220-L228). I personally think we should write tests for all these, though we'd obviously want to `skip = true` the specs if not running on Lucee.

### CODE REVIEW

File: models/EventHandler.cfc [hunk 1]
```diff
@@ -128,6 +128,61 @@ component extends="coldbox.system.remote.ColdboxProxy" implements="CFIDE.orm.IEv
 		);
 	}
 
+	/**
```
Comment (lmajano) [inline models/EventHandler.cfc:131]: All arguments should be lower cased camel case please.

**Review by lmajano — CHANGES_REQUESTED** [review]

**Comment by lmajano** [comment]
@michaelborn can we revisit this please

**Comment by lmajano** [comment]
@michaelborn can we revive this please?

**Comment by michaelborn** [comment]
Luis, I could use help on this Hibernate session error in my event handler test:

```
Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [User#8DFF91D7-0EB8-4530-8CF7A908C20124AD]
```

This appears to be some kind of concurrency issue, where an object is modified in the Hibernate session by some other transaction.

However... this doesn't make sense. I've created a totally NEW entity and there is no other code or threading which is modifying this entity.

Perhaps we can connect on this to resolve this issue. Most of the tests in the `EventHandlerTest.cfc` spec are failing with this error.

**Comment by lmajano** [comment]
Did you resolve it?

**Comment by michaelborn** [comment]
> Did you resolve it?

Yep, Jon Clausen helped me over the blocker. I was setting the entity `id` in my `entityNew()` call. Evidently that's not a valid way to create entities!

I'm now stuck on getting the new Hibernate events to trigger. I'll push up what I have so far.

**Comment by lmajano** [comment]
@michaelborn are you still working on this?

**Comment by michaelborn** [comment]
> @michaelborn ?

Apologies. I'll take another pass at the Lucee blockers.

**Comment by lmajano** [comment]
Is this ready for review again?

**Comment by michaelborn** [comment]
I'm going to close this PR. If the Lucee team ever decides to merge my PR which fixes these new Hibernate events, then we can take another look at this.

Sorry for the headache on this. It's just not worth the headspace.

See PR https://github.com/lucee/extension-hibernate/pull/31

**Pull request closed without merging**